src/azul/__init__.py

```diff
@@ -1157,8 +1162,22 @@ def _timing_is_restricted(self) -> bool:
         if self.lambda_context is None:
             return False
         else:
-            remaining = self.lambda_context.get_remaining_time_in_millis() / 1000
-            return remaining <= self.api_gateway_lambda_timeout
+            function_name = self.lambda_context.function_name
+            try:
+                app_name, _, _ = self.unqualified_resource_name_and_suffix(function_name)
+            except RequirementError:
+                from chalice.constants import (
+                    DEFAULT_HANDLER_NAME,
+                )
+                assert function_name == DEFAULT_HANDLER_NAME, function_name
+                return False
+            else:
+                if app_name == 'service':
+                    remaining = self.lambda_context.get_remaining_time_in_millis() / 1000
+                    return remaining <= self.api_gateway_lambda_timeout
+                else:
+                    assert app_name == 'indexer', (function_name, app_name)
+                    return False
```

Comment:
We may need to rethink this. First off, the timing is restricted in both REST APIs, that for the indexer and that for the service. Also, it is not immediately apparent to me when the RequirementError would be raised, so that definitely requires a comment for explanation, especially considering that the return value is False in that case.

Most importantly, I think we can better handle the distinction between REST API and other types invocations via Chalice middleware, potentially by injecting another property into the Config instance. The middleware that injects the lambda context is registered for `all` event types. We could register another middleware that is only registered for `http` events, which dynamically sets a boolean attribute in the Config instance to True. Timing is restricted whenever the attribute is True.

Reply:
> it is not immediately apparent to me when the RequirementError would be raised

The [default handler name](https://github.com/aws/chalice/blob/79838b02dc330cfe549823899d2662ded0538015/chalice/constants.py#L57) cannot be parsed by `Config.unqualified_resource_name_and_suffix` because it does not conform to our `azul-<deployment>-<resource>` naming convention. I don't remember when I observed this error but it did occur.

Reply:
> Most of the added code is shown as uncovered. I think the test needs to patch the Config property that enables/disables the DUOS stuff.

That property is only read when listing bundles so it has no effect on the test (which uses canned bundles).

I believe the uncovered methods (`list_bundles` and `fetch_bundle`) are already uncovered on `develop` because the SQL they use is incompatible with tinyquery (https://github.com/DataBiosphere/azul/issues/5046).

Reply:
Decided at PL to postpone test coverage.